lib/mix/lib/mix/utils.ex

```diff
@@ -15,28 +15,33 @@ defmodule Mix.Utils do
   stored there.
   """
   def mix_home do
-    mix_home_xdg_lookup("XDG_DATA_HOME")
+    mix_home_xdg_lookup(:user_data)
   end
 
   @doc """
   Gets possible location of global Mix configuration.
 
   Possible locations:
 
-   * `~/.mix`
-   * `MIX_HOME`
-   * `XDG_CONFIG_HOME/mix`
+     * `XDG_CONFIG_HOME/mix`
+     * `MIX_HOME`
+     * `~/.mix`
 
   """
   def mix_config do
-    mix_home_xdg_lookup("XDG_CONFIG_HOME")
+    mix_home_xdg_lookup(:user_config)
   end
 
   defp mix_home_xdg_lookup(xdg) do
-    case {System.get_env("MIX_HOME"), System.get_env(xdg)} do
-      {directory, _} when is_binary(directory) -> directory
-      {nil, directory} when is_binary(directory) -> Path.join(directory, "mix")
-      {nil, nil} -> Path.expand("~/.mix")
+    cond do
+      dir = System.get_env("MIX_HOME") ->
+        dir
+
+      System.get_env("MIX_XDG") ->
```

Comment:
Shouldn't this check for the value as well? Something along the lines of this:

```suggestion
      System.get_env("MIX_XDG") in ["1", "true"]->
```